projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx

```diff
@@ -0,0 +1,320 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { TokenInput } from "src/components/Swap/TokenInput";
+import { Token, TokenValue } from "@beanstalk/sdk";
+import { useAllTokensBalance } from "src/tokens/useTokenBalance";
+import styled from "styled-components";
+import { images } from "src/assets/images/tokens";
+import { useAccount } from "wagmi";
+import { ContractReceipt } from "ethers";
+import { Well } from "@beanstalk/sdk/Wells";
+import { useQuery } from "@tanstack/react-query";
+
+type RemoveLiquidityProps = {
+  well: Well;
+};
+
+enum REMOVE_LIQUIDITY_MODE {
+  Balanced,
+  OneToken,
+  Custom
+}
+
+type LiquidityAmounts = {
+  [key: number]: TokenValue;
+};
+
+export const RemoveLiquidity = ({ well }: RemoveLiquidityProps) => {
+  const { address } = useAccount();
+
+  const [wellTokens, setWellTokens] = useState<Token[]>();
+  const [wellLpToken, setWellLpToken] = useState<Token | null>(null);
+  const [lpTokenAmount, setLpTokenAmount] = useState<TokenValue | undefined>();
+  const [receipt, setReceipt] = useState<ContractReceipt | null>(null);
+  const [isLoadingAllBalances, setIsLoadingAllBalances] = useState(true);
+  const [removeLiquidityMode, setRemoveLiquidityMode] = useState<REMOVE_LIQUIDITY_MODE>(REMOVE_LIQUIDITY_MODE.Balanced);
+  const [singleTokenIndex, setSingleTokenIndex] = useState<number>(0);
+  const [amounts, setAmounts] = useState<LiquidityAmounts>({});
+
+  const { isLoading: isAllTokenLoading, refetch: refetchBalances } = useAllTokensBalance();
+
+  const atLeastOneAmountNonzero = useCallback(() => Object.values(amounts).filter((amount) => amount.value.gt("0")).length > 0, [amounts]);
+
+  useEffect(() => {
+    const fetching = isAllTokenLoading;
+    fetching ? setIsLoadingAllBalances(true) : setTimeout(() => setIsLoadingAllBalances(false), 500);
+  }, [isAllTokenLoading]);
+
+  useEffect(() => {
+    if (well.tokens) {
+      const tokens: Token[] = [];
+      const initialAmounts: LiquidityAmounts = {};
+      well.tokens.forEach((token, index) => {
+        token.setMetadata({ logo: images[token.symbol] ?? images.DEFAULT });
+        tokens.push(token);
+        initialAmounts[index] = TokenValue.ZERO;
+      });
+
+      setWellTokens(tokens);
+      setAmounts(initialAmounts);
+    }
+  }, [well]);
+
+  useEffect(() => {
+    if (well.lpToken) {
+      let lpTokenWithMetadata = well.lpToken;
+      lpTokenWithMetadata.setMetadata({ logo: images[well.lpToken.symbol] ?? images.DEFAULT });
```

Comment:
we should have a different default image for our Well LP tokens. cc @sweetredbeans

Reply:
I think it's great as is. :trollface: